Mlem/Views/Shared/AppLockView.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  AppLockView.swift
+//  Mlem
+//
+//  Created by Sumeet Gill on 2024-01-16.
+//
+
+import Dependencies
+import LocalAuthentication
+import SwiftUI
+
+struct AppLockView: View {
+    @Dependency(\.accountsTracker) var accountsTracker
+
+    @ObservedObject var biometricUnlock: BiometricUnlock
+    @Environment(\.scenePhase) var phase
+    
+    @AppStorage("appLock") var appLock: AppLock = .disabled
+    
+    @State var isErrorVisible: Bool = false
+    @State var isButtonHidden = true
+    @State var isAuthenticating = false
+    @State var alertMessage: String = ""
+
+    var body: some View {
+        @State var logoOffset: CGFloat = 0
+
+        VStack(spacing: 50) {
+            LogoView()
+                .frame(width: 150, height: 150)
+                .animation(.spring(), value: isAuthenticating)
+                .offset(y: isAuthenticating ? 0 : 300)
```

Comment:
I'm not a huge fan of the up-and-down animation. What do you think about a simple opacity animation instead?

```suggestion
        ZStack {
            LogoView()
                .frame(width: 150, height: 150)
                .opacity((isAuthenticating || biometricUnlock.isUnlocked) ? 0 : 1)
                .animation(.default, value: isAuthenticating || biometricUnlock.isUnlocked)
                .frame(maxHeight: .infinity, alignment: .center)
```

Reply:
![Simulator Screenshot - iPhone 15 Pro - 2024-01-18 at 13 02 02](https://github.com/mlemgroup/mlem/assets/1139817/b32245a6-6619-47ca-bfa5-a5a6f5bdfe54)


Maybe I copy pasted it in wrong but it doesn't seem to show the user anything when face ID is doing its challenge. Is that correct? 

I personally like the animation (obviously) because it lets the user know that they're in the Mlem app while touch/face ID is occuring.

Reply:
> Maybe I copy pasted it in wrong but it doesn't seem to show the user anything when face ID is doing its challenge. Is that correct?

That's correct, yes. For me, the FaceID animation is only shown for a second. I think the opacity-only animation looks cleaner, personally. It's more in-line with what I've seen before in other apps

Here's a video comparison:

https://github.com/mlemgroup/mlem/assets/78750526/c1816ba2-493e-4429-b58a-1ad78c9f4901

https://github.com/mlemgroup/mlem/assets/78750526/e7d905c7-d49b-4b8c-bf11-84c64e8c4253


Another thought I had - the animation for FaceID incorporates the dynamic island for devices that have it. This means that the animation isn't in the middle anymore. I think we could always show the logo in the middle whilst FaceID is unlocking in this case?

Reply:
Check out my latest changes, it's a hybrid between both of ours. If it has dynamic island then it does your changes, which is fades logo out. Otherwise it moves the logo up slightly. Does that work?

Reply:
Sure, I'll check it out when I get home later. 

> If it has Dynamic Island it does your changes

Do we need to animate the logo at all when there's a Dynamic Island? We don't need to move it out of the way of the animation - I haven't tested it yet, but I think we should just be able to leave the logo at 100% opacity in the center rather than animating it.

> Also something to note: `.ignoresSafeArea()` seems to push the button below the bottom bar for iPhone 15 pro.

I put a bottom padding on the button in my review comment suggestion to mitigate this. Sorry, I should have mentioned it 😄

Reply:
It doesn't move out of the way if there's Dynamic Island right now. I have it fading in and out of opacity but that wouldn't be needed either so I can remove that part. 


I'll add padding to the button.

Reply:
Hmm, I still prefer the opacity-only animation over the bouncy one - I think the unlock happens too quickly to justify making the logo move. I've used a few apps that have tried various approaches to this, and I've always preferred the animations that are simple and smooth (like Apollo) vs ones with more animation (like Signal). In my opinion we should just use the simpler opacity-only animation